### DISCUSSION TITLE
Controller not found - Error 500

### DISCUSSION THREAD

**Discussion by hallleron** [pimcore/pimcore#9497] [General]
Hey,

I am new to Pimcore and decided to create an own bundle. Unfortunately, my controller (Controller/IndexController.php) with the namespace `Mycompany\AddressBookBundle\Controller\IndexController` doesn't get recognized with the following route configuration in routing.yaml:

```
mycompany_address_book_homepage:
    path:     /
    defaults: { _controller: MycompanyAddressBookBundle:Index:index }
```

The controller also has the required `indexAction()` function and the bundle itself was generated by using the https://github.com/pimcore/bundle-generator. I am really stuck here. Anyone have an idea?

<img width="1041" alt="Bildschirmfoto 2021-06-18 um 14 46 35" src="https://user-images.githubusercontent.com/9599650/122563020-0084f880-d044-11eb-906f-cbf3cdd66254.png">

**Comment by dvesh3** [comment]
@hallleron if you are facing error "XYZController has no container set, did you forget to define it as a service subscriber?" then please define controllers as a service in bundle Resources/config/services.yml:
```
    Mycompany\AddressBookBundle\Controller\:
        resource: '../../Controller'
        public: true
        tags: ['controller.service_arguments']
```

**Reply by hallleron** [reply]
> @hallleron if you are facing error "XYZController has no container set, did you forget to define it as a service subscriber?" then please define controllers as a service in bundle Resources/config/services.yml:
> 
> ```
>     Mycompany\AddressBookBundle\Controller\:
>         resource: '../../Controller'
>         public: true
>         tags: ['controller.service_arguments']
> ```

Hey,

thank you for the quick answer! I checked my file and this is already defined there. Exactly the same as you wrote.

My controller code looks like this:

```
<?php

namespace Mycompany\AddressBookBundle\Controller;

use Pimcore\Controller\FrontendController;
use Symfony\Component\HttpFoundation\Request;
use Symfony\Component\HttpFoundation\Response;

class IndexController extends FrontendController
{

    public function indexAction(Request $request): Response
    {
        return new Response('Hello world from abcpremium_address_book');
    }
}
```

**Reply by dvesh3** [reply]
try changing your route config:
```
defaults: { _controller: Mycompany\AddressBookBundle\Controller\IndexController:indexAction }
```

**Reply by hallleron** [reply]
> Mycompany\AddressBookBundle\Controller\IndexController:indexAction

This worked! Thank you very much! But why can't I use the other notation?

**Reply by dvesh3** [reply]
Symfony removed the support on version 5.
